### PR TITLE
Hsm key rotation

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

**Comment by VaclavK** [comment]
FYI - @DerekGn

File: src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            public string MasterSecretName { get; } = "MASTERSECRET";
+
+            [Option(
+                Description = "key strength of master key (defaults to 2048) - possible values are 2048, 3072, 4096",
+                ShortName = "c",
+                LongName = "masterKeyStrength",
+                ShowInHelpText = true)]
+            public int MasterKeyStrength { get; } = 2048;
+
+            private static readonly int[] AllowedKeyStrengths = {2048, 3072, 4096};
```
Comment (DerekGn) [inline src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs]: Have you tried changing the master key size once it has been created as a HSM key not sure if that can be done.

File: src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            // ReSharper disable once MemberCanBePrivate.Global
+            public string KeyVaultName { get; set; }
+
+            [Option(
+                Description = "name of master HSM backed key (defaults to 'RSAHSMKEY')",
+                ShortName = "a",
+                LongName = "masterKeyName",
+                ShowInHelpText = true)]
+            // ReSharper disable once StringLiteralTypo
+            public string MasterKeyName { get; } = "RSAHSMKEY";
```
Comment (DerekGn) [inline src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs]: The default in the secure data store config is MASTERKEY, I used RSAHSMKEY in the POC but changed it to MASTERKEY in the release version.

File: src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs (hunk 1 above)
Comment (VaclavK) [reply src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs]: yeah that works

File: src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs (hunk 2 above)
Comment (VaclavK) [reply src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs]: ok will ammend

File: src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs (hunk 1 above)
Comment (DerekGn) [reply src/EShopWorld.Tools/Commands/Security/SecurityCommand.cs]: ok cool

**Comment by VaclavK** [comment]
@djfr FYI
@DerekGn few tweaks+test layer but the core is as before...

**Pull request merged**